src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataReadMeControl.xaml.cs

```diff
@@ -19,11 +15,84 @@ namespace NuGet.PackageManagement.UI
     /// <summary>
     /// Interaction logic for PackageMetadataReadMeControl.xaml
     /// </summary>
-    public partial class PackageMetadataReadMeControl : UserControl
+    public partial class PackageMetadataReadMeControl : UserControl, IDisposable
     {
+#pragma warning disable CS0618 // Type or member is obsolete
+        private IMarkdownPreview _markdownPreview;
+#pragma warning restore CS0618 // Type or member is obsolete
+        private bool _disposed = false;
+
+        [Obsolete]
         public PackageMetadataReadMeControl()
         {
             InitializeComponent();
         }
+
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                _markdownPreview?.Dispose();
+                var viewModel = (ReadMePreviewViewModel)DataContext;
+                viewModel.PropertyChanged -= ViewModel_PropertyChangedAsync;
```

Comment:
This will only unregister the event handler for the last viewModel when the PMRMC disposes. You should instead make sure this unregister happens when the viewModel changes, unregister the old viewModel, and register the new one.

Reply:
Okay, added it to the context change and unregistered it here too, needs to be removed in both instances right?

Reply:
Yeah, you're right. It should also be unregistered in the dispose.